lib/metrics.ts

```diff
@@ -0,0 +1,42 @@
+import * as _ from 'lodash';
+
+interface Metrics {
+  fetchCallGraphBuilder?: number;
+  getMvnClassPath?: number;
+  getGradleClassPath?: number;
+  getEntrypoints?: number;
+  generateCallGraph?: number;
+  mapClassesPerJar?: number;
+  getCallGraph?: number;
+}
+
+const metricsState: { [x in keyof Metrics]: [number, number] | undefined } = {};
+
+function start(label: keyof Metrics) {
+  metricsState[label] = process.hrtime();
+}
+
+function stop(label: keyof Metrics) {
+  const previous = metricsState[label];
+  metricsState[label] = process.hrtime(previous);
+}
+
+function getMetrics(): Metrics {
+  return _(metricsState)
+    .mapValues<number>(
+      (hrtime) => _.get(hrtime, '[0]', 0) + _.get(hrtime, '[1]', 0) / 1e9,
```

Comment:
Don't you need here parenthesis? e.g.
`(_.get(hrtime, '[0]', 0) + _.get(hrtime, '[1]', 0)) / 1e9,`

Reply:
No, the second item from hrtime, is `nanoseconds`

Reply:
Oh, I see,  any benefit of doing it here and not on the `stop` function?

Reply:
just to keep the actual result unchanged, and modify as needed when presenting.